src/map/lua/lua_baseentity.cpp

```diff
@@ -2732,6 +2732,55 @@ inline int32 CLuaBaseEntity::resetPlayer(lua_State *L)
     return 1;
 }
 
+/************************************************************************
+*  Function: goToEntity()
+*  Purpose : Transports PC to a Mob or NPC; works across multiple servers
+*  Example : player:goToEntity(ID, Option)
+*  Notes   : Option 0: Spawned/Unspawned | Option 1: Spawned only
+************************************************************************/
+
+int32 CLuaBaseEntity::goToEntity(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
+
+    bool warped = true;
+
+    if (!lua_isnil(L, 1) && lua_isnumber(L, 1))
+    {
+
+        CCharEntity* PChar = (CCharEntity*)m_PBaseEntity;
+
+        bool spawnedOnly  = !lua_isnil(L, 2) ? lua_tonumber(L, 2) : 0;
+
+        uint32 targetID   = (uint32)lua_tonumber(L, 1);
+        uint16 targetZone = (targetID >> 12) & 0x0FFF;
+        uint16 playerID   = m_PBaseEntity->id;
+        uint16 playerZone = PChar->loc.zone->GetID();
+
+        char buf[12];
+        memset(&buf[0], 0, sizeof(buf));
+
+        ref<bool>  (&buf,  0) = true; // Toggle for message routing; goes to entity server first
+        ref<bool>  (&buf,  1) = spawnedOnly; // Specification for Spawned Only or Any 
+        ref<uint16>(&buf,  2) = targetZone;
+        ref<uint16>(&buf,  4) = playerZone;
+        ref<uint32>(&buf,  6) = targetID;
+        ref<uint16>(&buf, 10) = playerID;
+
+        message::send(MSG_SEND_TO_ENTITY, &buf[0], sizeof(buf), nullptr);
+
+        if (PChar->status != STATUS_DISAPPEAR)
```

Comment:
this is technically a race condition: message::send does not block, so you have no way of knowing whether the message went through yet.  It's probably better to just return nothing since it can't realistically be able to determine if you are warping or not